### ISSUE TITLE
platform.linux_distribution is deprecated in python 3.7 and removed in 3.8

### ISSUE THREAD

**Issue by mp15** [irods/irods#4951]
### iRODS Version, OS and Version
4.2.7

### What did you try to do?
CMakeLists.txt contains a reference to platform.linux_distribution which no longer exists this causes an error when running cmake.

### Expected behavior
IRODS uses a non-deprecated method to check linux_distribution.

### Observed behavior (including steps to reproduce, if applicable)

```
-- Found OpenSSL: $PREFIX/lib/libcrypto.so (found version "1.0.2o") 
-- Found pam: /usr/lib/x86_64-linux-gnu/libpam.so
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: module 'platform' has no attribute 'linux_distribution'
CMake Error at CMakeLists.txt:93 (message):
  Linux platform name detection failed

  1


-- Configuring incomplete, errors occurred!
See also "$SRC_DIR/build/CMakeFiles/CMakeOutput.log".
See also "$SRC_DIR/build/CMakeFiles/CMakeError.log".
Traceback (most recent call last):
  File "/home/gsinstall/miniconda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/home/gsinstall/miniconda/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 474, in main
    execute(sys.argv[1:])
  File "/home/gsinstall/miniconda/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 465, in execute
    verify=args.verify, variants=args.variants)
  File "/home/gsinstall/miniconda/lib/python2.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/home/gsinstall/miniconda/lib/python2.7/site-packages/conda_build/build.py", line 2863, in build_tree
    notest=notest,
  File "/home/gsinstall/miniconda/lib/python2.7/site-packages/conda_build/build.py", line 1994, in build
    cwd=src_dir, stats=build_stats)
  File "/home/gsinstall/miniconda/lib/python2.7/site-packages/conda_build/utils.py", line 404, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/home/gsinstall/miniconda/lib/python2.7/site-packages/conda_build/utils.py", line 384, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/home/gsinstall/miniconda/conda-bld/irods_1590711835372/work/conda_build.sh']' returned non-zero exit status 1
```

**Comment by mp15** [comment]
I would recommend we replace this with a `lsb_release -a` call as is used elsewhere in the CMakeList.txt.

**Comment by trel** [comment]
this is a subset of #2037

**Comment by korydraughn** [comment]
Resolved by 9215fb2204df1050b2a97f602ba839ea9c36d3c5
